### PR TITLE
Relations, life cycle callbacks, fixtures

### PR DESCRIPTION
**Pull request by gconstantin**

### CODE REVIEW

File: src/Entity/Affiliate.php [hunk 1]
```diff
@@ -151,10 +168,52 @@ public function getCreatedAt(): ?\DateTime
...
+
+        return $this;
+    }
+
+    /**
+     * @ORM\PrePersist()
+     */
+    public function prePersist(): void
+    {
+        $this->createdAt = new \DateTime('now', new\DateTimeZone('UTC'));
```
Comment (gregurco) [inline src/Entity/Affiliate.php]: usually timezone is setup on the level of php config. No need to use second param. Also the first one is default. You can write simply: `$this->createdAt = new \DateTime();`

File: src/Repository/AffiliateRepository.php [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<?php
+
+namespace App\Repository;
+
+use Doctrine\ORM\EntityRepository;
+
+
+class AffiliateRepository extends EntityRepository
```
Comment (gregurco) [inline src/Repository/AffiliateRepository.php]: see here how repository should looks like: https://symfony.com/doc/current/doctrine.html#querying-for-objects-the-repository
You did in in sf 3.3 style.

**Comment by gconstantin** [comment]
Added changes

File: src/DataFixtures/CategoryFixtures.php [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        return static::generateReferenceKey(\rand(1, CategoryFixtures::COUNT));
+    }
+
+    /**
+     * @param int $id
+     * @return string
+     */
+    public static function generateReferenceKey($id): string
+    {
+        return CategoryFixtures::REFERENCE.$id;
```
Comment (gregurco) [inline src/DataFixtures/CategoryFixtures.php]: code style fix, should be:
`return CategoryFixtures::REFERENCE . $id;`

File: src/DataFixtures/CategoryFixtures.php [hunk 4]
```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace App\DataFixtures;
+
+use Doctrine\Bundle\FixturesBundle\Fixture;
+use Doctrine\Common\Persistence\ObjectManager;
+use Faker\Factory;
```
Comment (gregurco) [inline src/DataFixtures/CategoryFixtures.php]: import `use Faker;` and use it as `$faker = Faker\Factory::create();`

**Review by gregurco — CHANGES_REQUESTED** [review]

**Comment by gconstantin** [comment]
Fixed

**Comment by gregurco** [comment]
@gconstantin :+1:

**Pull request merged**